packages/components/src/panel/body.tsx

```diff
@@ -30,27 +36,29 @@ export function PanelBody(
 		opened,
 		title,
 		scrollAfterOpen = true,
-	},
-	ref
-) {
-	const [ isOpened, setIsOpened ] = useControlledState( opened, {
-		initial: initialOpen === undefined ? true : initialOpen,
-	} );
-	const nodeRef = useRef();
+	} = props;
+	const [ isOpened, setIsOpened ] = useControlledState< boolean | undefined >(
+		opened,
+		{
+			initial: initialOpen === undefined ? true : initialOpen,
+			fallback: false,
+		}
+	);
+	const nodeRef = useRef< HTMLElement >( null );
 
 	// Defaults to 'smooth' scrolling
 	// https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView
 	const scrollBehavior = useReducedMotion() ? 'auto' : 'smooth';
 
-	const handleOnToggle = ( event ) => {
+	const handleOnToggle = ( event: React.MouseEvent ) => {
 		event.preventDefault();
 		const next = ! isOpened;
 		setIsOpened( next );
 		onToggle( next );
 	};
 
 	// Ref is used so that the effect does not re-run upon scrollAfterOpen changing value.
-	const scrollAfterOpenRef = useRef();
+	const scrollAfterOpenRef = useRef< boolean | null >( null );
```

Comment:
Yes, it is clear. Thanks for explaining about `nodeRef` because I would have had that question. 😄  

And just to be certain I fully understand - the only reason we would want to switch it to `undefined` in this case is to avoid introducing a runtime change?

Reply:
> the only reason we would want to switch it to undefined in this case is to avoid introducing a runtime change?

That's definitely the main reason, yeah.

If I had to write this piece of code from scratch, I would have assigned a default value of `false`, instead of `undefined` — that way, the type associated to `useRef` would only be `boolean`, instead of `boolean | undefined`.

In general, I always try to avoid mixing-and-matching types when possible — unless a certain type has a clear meaning (like `null` in the case of `nodeRef`).

Also, writing `useRef( undefined )` is just a more explicit version of `useRef()` — and therefore, you could change the code to a less verbose

```ts
const scrollAfterOpenRef = useRef< boolean | undefined >();
```

Reply:
Thanks, this is helpful!